Documentation/CraftingNuGetPackages.md

```diff
@@ -0,0 +1,69 @@
+# Crafting a NuGet Package for MSB4U
+
+> MSBuildForUnity consumes .NET based packages without any modifications to the package. This documentation describes how C++, WinRT or packages with Unity-specific contents can easily be adapted to work with MSBuildForUnity.
+
+## Quickstart Summary
+
+Though the use of a [Build Targets or Props](https://docs.microsoft.com/en-us/nuget/create-packages/creating-a-package#include-msbuild-props-and-targets-in-a-package) file inside a NuGet package, we can leverage properties provided by MSBuildForUnity to appropriately select the contents to bring into the Unity project. The following snippet is taken from [Microsoft.Windows.MixedReality.DotNetWinRT](https://www.nuget.org/packages/Microsoft.Windows.MixedReality.DotNetWinRT/) NuGet package. It brings in the `Unity` folder packaged in the root of the NuGet package to the output folder inside Unity.
+
+```xml
+<Project>
+  <PropertyGroup>
+    <!--All extra resources for this package will be placed under this folder in the output directory.-->
+    <PackageDestinationFolder>$(MSBuildThisFileName)</PackageDestinationFolder>
+  </PropertyGroup>
+  
+  <ItemGroup Condition="'$(MSBuildForUnityVersion)' != ''">
+    <Content Include="$(MSBuildThisFileDirectory)..\Unity\**">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+      <!-- Don't show .meta files in Solution Explorer - it's not useful. -->
+      <Visible Condition="'%(Extension)' == '.meta'">false</Visible>
+      <Link>$(PackageDestinationFolder)\%(RecursiveDir)%(Filename)%(Extension)</Link>
+    </Content>
+  </ItemGroup>
+</Project>
+```
+
+## Understanding MSBuildForUnity Behavior
+
+MSBuildForUnity is based on MSBuild and thereby leverages the build rules including package and project reference resoluton of MSBuild. These steps are invoked at Editor Compilation stage with a set of MSBuildForUnity provided variables. At this stage, we have the following information available:
```

Comment:
nit: its not super clear why this information is relevant based on the preceding sections. A sentence should exist prior to calling out what you know/dont know that touches on the reason for needing to know this.

It looks like the final sentence calls out that platform specific dlls need custom work. It seems something explaining that need/problem should be in this paragraph.

Reply:
+1 to a little confusion here - since we're using specific terms (i.e. Editor Compilation) it would be helpful to link to other resources that describe exactly what they are. i.e. is Editor Compilation == Unity editor compilation? Does it mean some other sort of editor compilation (i.e. VS?)